### PR TITLE
Document public facing API

### PR DESCRIPTION
**Pull request by JerrettDavis**
Documented the publicly facing API.

#1071 

Can we tackle #1072 in here as well? Return types can be part of the docs as well, would like to add that in one go.

### CODE REVIEW

**Comment by JerrettDavis** [comment]
I have documented the entirety of the public API and the configuration. I have also added return types to all applicable fields and methods.

**Review by FabianGosebrink — APPROVED** [review]

File: projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts [hunk 1]
```diff
@@ -1,43 +1,147 @@
...
   clientId?: string;
+  /**
+   * 'code', 'id_token token' or 'id_token' Name of the flow which can be configured.
+   * You must use the 'id_token token' flow, if you want to access an API or get user data from the server.
+   * The access_token is required for this, and only returned with this flow.
+   */
   responseType?: string;
+  /**
+   * This is this scopes which are requested from the server from this client.
+   * This must match the STS server configuration.
```
Comment (damienbod) [inline projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts:28]: Can you append this:

The openid scope is required. This offline_access scope can be requested when using refresh tokens but this is optional and some STS do not support this or recommend not requesting this even when using refresh tokens in the browser.

File: projects/angular-auth-oidc-client/src/lib/oidc.security.service.ts [hunk 2]
```diff
@@ -58,96 +76,164 @@ export class OidcSecurityService {
...
+   * https://tools.ietf.org/html/rfc7009
+   *
+   * @param refreshToken The access token to revoke.
+   */
+  revokeRefreshToken(refreshToken?: any): Observable<any> {
     return this.logoffRevocationService.revokeRefreshToken(refreshToken);
   }
 
+  /**
+   * Creates the ENS session URL which can be used to implement an alternate server logout.
```
Comment (damienbod) [inline projects/angular-auth-oidc-client/src/lib/oidc.security.service.ts:235]: ENS => END

File: projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts [hunk 3]
```diff
@@ -1,43 +1,147 @@
...
+  /**
+   * Activates Pushed Authorisation Requests for login and popup login.
+   * Not compatible with Iframe renew.
+   */
   usePushedAuthorisationRequests?: boolean;
+  /**
+   * 	A token obtained by using a refresh token normally doesn't contain a nonce value.
+   * 	However, some OIDC endpoint implementations do send one. The library checks to see if the nonce is present.
+   * 	Note that if the nonce value is present, it will not be verified.
+   * 	Default value is false
```
Comment (damienbod) [inline projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts:69]: append

This is not recommended, if the STS returns a nonce in the refresh.

**Comment by damienbod** [comment]
@JerrettDavis LGTM, just 3 comments

Greetings Damien

File: projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts (hunk 3 above)
Comment (JerrettDavis) [reply projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts]: This is now corrected.

File: projects/angular-auth-oidc-client/src/lib/oidc.security.service.ts (hunk 2 above)
Comment (JerrettDavis) [reply projects/angular-auth-oidc-client/src/lib/oidc.security.service.ts]: This is now corrected.

File: projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts (hunk 1 above)
Comment (JerrettDavis) [reply projects/angular-auth-oidc-client/src/lib/config/openid-configuration.ts:28]: This is now corrected.

**Pull request merged**